src/coreclr/jit/emitarm64.cpp

```diff
@@ -13481,8 +13481,24 @@ void emitter::emitInsLoadStoreOp(instruction ins, emitAttr attr, regNumber dataR
                 }
                 else // no scale
                 {
-                    // Then load/store dataReg from/to [memBase + index]
-                    emitIns_R_R_R(ins, attr, dataReg, memBase->GetRegNum(), index->GetRegNum());
+                    if (index->OperIs(GT_BFIZ) && index->isContained())
+                    {
+                        // Then load/store dataReg from/to [memBase + index*scale with sign/zero extension]
+                        GenTreeCast* cast = index->gtGetOp1()->AsCast();
+
+                        // For now, this code only supports extensions from i32/u32
+                        assert(cast->isContained() && varTypeIsInt(cast->CastFromType()));
+
+                        const bool isZeroExtended = cast->IsUnsigned() || varTypeIsUnsigned(cast->CastToType());
```

Comment:
This should be just `cast->IsUnsigned()` because for `int -> long` that's what decides whether the cast is zero-extending or not. There are casts that have `TYP_ULONG` as the target type but no `GTF_UNSIGNED` set, they are sign-extending.

Reply:
hm, good point!